src/main/scala/Piece.scala

```diff
@@ -21,11 +21,9 @@ case class Piece(color: Color, role: Role) {
       case Rook   => from onSameLine to
       case Bishop => from onSameDiagonal to
       case Knight =>
-        from.color != to.color && {
```

Comment:
This check is not necessary, so it may have been a micro-optimization. No performance regression in perft test.